### DISCUSSION TITLE
Expanding Activation Functions Module

### DISCUSSION THREAD

**Discussion by jrego43** [LazaroHurtado/micrograd-rs#5] [Ideas]
Activation functions are the key to representing complex nonlinear functions with neural networks. As such, it is important that we implement a wide variety of them to ensure a flexible framework. You can see many activation functions [here](https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity) that should be considered for implementation. As of now, we have **ReLU**, **Softmax**, and **Tanh** completed, so you can use their source files for reference.

**Comment by jrego43** [comment]
~I'll be working on the sigmoid activation function starting tomorrow.~

⬇️

**Reply by Pochetes** [reply]
update from signal: @jrego43 will be working on #3 and I'll be doing this one. slight confusion there
